### PR TITLE
task/4119/dcl_cms_plugins

### PR DESCRIPTION
**Pull request by tiwariav**
CMS plugins for current DCL pages.
(_Before merging, existing cms pages using the dcl template must be re-written._)

All the new plugins for the DCL are listed below the header **Datacenterlight** in the plugins list.

## The plugins are:
- **DCLNavbarPlugin**
    - This plugin is added by default to any DCL page. By default it contains the logo and language options. Every other link has to be added as a new child plugin **DCLLinkPlugin** or **DCLDropdownPlugin**
- **DCLLinkPlugin**
    - Used to add a link to the navbar of footer of the page. It can only be a child of **DCLNavbarPlugin**, **DCLDropdownPlugin** and **DCLFooterPlugin**.
      the **_target_** field for the plugin can be a _valid url_, or the _id of another section from the page_.
- **DCLNavbarDropdownPlugin**
    - Used to add a dropdown to **DCLNavbarPlugin**. Can also have an optional **_target_** to scroll to when clicked.
- **DCLFooterPlugin**
    - Added by default to a page, it contains copyright text and can contain **DCLLinkPlugin** for links.
- **DCLSectionPlugin**
    - Adds a new dcl section to the body of a page. Along with a **_heading_** and an html **_content_**. Other fields for the plugin are:
**_text_direction_** - to align the section _left_ or _right_.
**_html_id_** - id to be set in the rendered html. Can be used in the _target_ field of **DCLLinkPlugin** item.
**_plain_heading_** - style the section with whydcl style (no border for heading, thinner image section)
**_center_on_mobile_** - whether to center the text on mobile or not.
**_background_gradient_** - whether to add a background_gradient.
- **DCLCalculatorPlugin**
    - Added by default to a page. It includes the calculator along with other options of **DCLSectionPlugin**
- **DCLSectionIconPlugin**
    - Adds a fontawesome icon to the half of a **DCLSectionPlugin**.
- **DCLSectionIconPlugin**
    - Adds one or more images to the half of a **DCLSectionPlugin**. Can also contain an optional **_caption_** for each image.
      _If a single image is added_, it will be center aligned and stretch till its max dimensions or shrink if required.
      _If a multiple images are added_, there will two images in a row for screens wider than 420px and one for smaller. If the last row has a single image, it will stretch and center align to center.
- **DCLBannerListPlugin**
    - Adds a section with list of **DCLBannerItemPlugin** with a **_heading_**.
- **DCLBannerItemPlugin**
    - Adds a banner item with **_content_** on one side and styled **_banner_text_** and **_banner_image_** on other (as on the whydcl page)
**_text_direction_** - to align the section _left_ or _right_.

### CODE REVIEW

**Comment by verysanghee** [comment]
great job! some of things I noticed 
- calculator OS doesn't work.
- missing Tech Stack plugin. 
- footer stlye is slightly off.

**Comment by pcoder** [comment]
Waiting for above points to be done before I review ;)

**Comment by verysanghee** [comment]
above points nicely fixed now :D

File: hosting/templates/hosting/calculator_form.html [hunk 1]
```diff
@@ -71,15 +71,6 @@
                     {% endfor %}
                 </select>
             </div>
-            <div class="help-block with-errors">
```
Comment (pcoder) [inline hosting/templates/hosting/calculator_form.html]: Could we do this in a separate PR? Does not make sense to me to something unrelated,

File: datacenterlight/views.py [hunk 2]
```diff
@@ -112,37 +112,41 @@ def post(self, request):
...
         except ValidationError as err:
             msg = '{} : {}.'.format(cores, str(err))
-            messages.add_message(self.request, messages.ERROR, msg,
-                                 extra_tags='cores')
-            return HttpResponseRedirect(
-                reverse('datacenterlight:index') + "#order_form")
+            messages.add_message(
+                self.request, messages.ERROR, msg, extra_tags='cores'
+            )
+            return HttpResponseRedirect(referer_url + "#order_form")
```
[the remark this answers is not in the document]
Comment (tiwariav) [reply datacenterlight/views.py]: Sorry, but it's a little unclear to me, what info are we concerned to store for multiple domains?

File: hosting/templates/hosting/calculator_form.html (hunk 1 above)
Comment (tiwariav) [reply hosting/templates/hosting/calculator_form.html]: Sure. Its just a duplicate block of code. Can be removed in any PR concerned with the calculator.

File: dynamicweb/settings/base.py [hunk 3]
```diff
@@ -227,7 +227,7 @@ def int_env(val, default_value=0):
     ('blog_ungleich.html', gettext('Blog')),
     ('page.html', gettext('Page')),
     # dcl
-    ('datacenterlight/cms_page.html', gettext('Data Center Light')),
```
[the remark this answers is not in the document]
Comment (tiwariav) [reply dynamicweb/settings/base.py]: Done 👍

**Comment by tiwariav** [comment]
Fixed the unnecessary tags loaded in templates.
